packages/core/src/shorthands/border.ts

```diff
@@ -29,15 +31,28 @@ export function border(width: BorderWidthInput, style: BorderStyleInput, color:
  *
  * @example
  *  border('2px')
+ *  border('solid')
  *  border('2px', 'solid')
+ *  border('solid', '2px')
  *  border('2px', 'solid', 'red')
+ *  border('solid', '2px', 'red')
  *
  * See https://developer.mozilla.org/en-US/docs/Web/CSS/border
  */
-export function border(...values: [BorderWidthInput, BorderStyleInput?, BorderColorInput?]): BorderStyle {
+export function border(
```

Comment:
does it make more sense to use some kind of runtime check to determine the values?

There is no point IMO is 'half' supporting variadic arguments since the CSS spec states that they can be in any order

https://developer.mozilla.org/en-US/docs/Web/CSS/border#syntax